Data generation script, initial front end set-up (webpack), express service set up

jlukenoff opened the pull request.

DavidPatterson-Cole commented:
Overall looks really good. A few small points: 

1.  I would recommend adding a comment to explain what you did and what specifically you want 
reviewed - there is a good example of a PR comment in learn
2. In your server file you require the calendarGenerator and invoke it immediately (which will keep adding data to your database on each server restart). I haven't used middleware extensively by any means but it might be more correct to put that in a middleware function.
3. I really like the example you commented in your dataGenerator file
4. It would be helpful to have a comment at the top of each function saying what it is doing (e.g., for generateDates it would be great to have a comment that specifies these are dates that have already been booked)

jlukenoff commented:
Thanks a lot for the feedback David!

- only meaningful changes occur in server and database folder rest is tldr;
- Updated README to be more descriptive, include start up instructions
- removed data generation from server script to npm script to only be run explicitly by the developer
